build-common/src/org/jetbrains/kotlin/incremental/IncrementalJvmCache.kt

```diff
@@ -439,7 +432,7 @@ open class IncrementalJvmCache(
 
     // todo: reuse code with InlineFunctionsMap?
     private inner class ConstantsMap(storageFile: File) :
-        BasicStringMap<LinkedHashMap<String, Any>>(storageFile, LinkedHashMapExternalizer(StringExternalizer, ConstantExternalizer)) {
+        BasicStringMap<Map<String, Any>>(storageFile, MapExternalizer(StringExternalizer, ConstantValueExternalizer)) {
```

Comment:
What is the benefit of replacing Map with LinkedHashMap? Previously, it was clear that this was an ordered collection.

Reply:
Right, I also liked the fact that it was ordered, but I made this change because:
   1. `LinkedHashMap` feels a bit heavy.
   2. All collections in Kotlin are already implicitly ordered.
   3. If we really want to preserver order, having it a `LinkedHashMap` here is not enough. We will have to preserve order all the way from source to sink, and we would want to do that for all other data structures as well because almost everything involved in build should ideally be ordered. So we will end up turning all `Map`s and `Set`s throughout the codebase into `LinkedHashMap` and `LinkedHashSet`.
        - Actually, when implementing the new IC, I initially wrote new code with `LinkedHashMap` but over time found that it doesn't scale and the code just looks inconsistent if sometimes we use `Map`, sometimes we use `LinkedHashMap`. (Here is our previous discussion on this: https://github.com/google/kotlin/commit/654c4250ba81ce0d02c167c39de96b1212b3c2e7#commitcomment-57900126)

That said, I can also revert to `LinkedHashMap` if you think it's still nice to have.

Reply:
kk, let's keep it as-is.